cplusplus/VImage.cpp

```diff
@@ -189,7 +189,7 @@ VOption::set( const char *name, const char *value )
 
 // input vips object (image, source, target, etc. etc.)
 VOption *
-VOption::set( const char *name, const VObject value )
+VOption::set( const char *name, const VObject& value )
```

Comment:
`VObject` is designed to be a single pointer, so there's no performance advantage to passing by reference. 

Plus this would change the ABI, though perhaps that's not so important for C++.

Reply:
I think that `clang-tidy`'s [`performance-unnecessary-value-param`](https://clang.llvm.org/extra/clang-tidy/checks/performance-unnecessary-value-param.html) check is indeed wrong here. Somehow it detects that `VObject` (and derivatives) is expensive to copy. If you remove the `VObject` destructor (`~VObject()`) and copy constructor
(`VObject( const VObject &a ) : vobject( a.vobject )`) these warnings will disappear.

Reply:
Perhaps pass by reference would avoid the ref/unref pair? It still changes the ABI though.